optimade/server/main.py

```diff
@@ -70,32 +70,26 @@ def load_entries(endpoint_name: str, endpoint_collection: MongoCollection):
 app.add_exception_handler(NotImplementedError, exc_handlers.not_implemented_handler)
 app.add_exception_handler(Exception, exc_handlers.general_exception_handler)
 
-
-# Add various endpoints to `/vMAJOR`
-app.include_router(info.router, prefix=BASE_URL_PREFIXES["major"])
-app.include_router(links.router, prefix=BASE_URL_PREFIXES["major"])
-app.include_router(references.router, prefix=BASE_URL_PREFIXES["major"])
-app.include_router(structures.router, prefix=BASE_URL_PREFIXES["major"])
-
-
-# Add the router for the landing page for all prefixes
-app.include_router(landing.router)
-app.include_router(landing.router, prefix=BASE_URL_PREFIXES["major"])
+# Add various endpoints to unversioned URL
+for endpoint in (info, links, references, structures, landing, versions):
+    app.include_router(endpoint.router)
+    app.include_router(links.router)
+    app.include_router(references.router)
+    app.include_router(structures.router)
+    app.include_router(landing.router)
 
 
 def add_optional_versioned_base_urls(app: FastAPI):
     """Add the following OPTIONAL prefixes/base URLs to server:
     ```
+        /vMajor
         /vMajor.Minor
         /vMajor.Minor.Patch
     ```
     """
-    for version in ("minor", "patch"):
-        app.include_router(info.router, prefix=BASE_URL_PREFIXES[version])
-        app.include_router(links.router, prefix=BASE_URL_PREFIXES[version])
-        app.include_router(references.router, prefix=BASE_URL_PREFIXES[version])
-        app.include_router(structures.router, prefix=BASE_URL_PREFIXES[version])
-        app.include_router(landing.router, prefix=BASE_URL_PREFIXES[version])
+    for version in ("major", "minor", "patch"):
+        for endpoint in (info, links, references, structures, landing):
+            app.include_router(endpoint.router, prefix=BASE_URL_PREFIXES[version])
```

Comment:
Why not get rid of this? The main implementation doesn't have to implement all the optional things in the spec.

Reply:
Sure. But it should at the same time showcase how it may be done. And in some ways also be a "role model" for an implementation.
In my opinion the versioned base URLs, while OPTIONAL, are quite nice to have.

Reply:
Maybe just the major then. I feel like we're trying too hard to be a complete implementation with ALL the bell and whistles when they may not be relevant to many implementations. If versions are likely 3-4 years apart, then not having versioned base URLs is unlikely to much effect most implementations.

Reply:
The 3-4 years was a guess (although probably an accurate one).
Just having the major version would be a fine compromise for me.
But then I may ask, if we're doing the major version, why not also add the minor and patch/full version?

Reply:
Because it takes away from API clarity. At some point, we put in too much redundancy and obfuscate what the API does.

Reply:
Just to weigh in here, I would agree with you @shyamd, if the versioned URLs were left in the OpenAPI schema. This is no longer the case, and instead only references to the unversioned endpoints appear there. I don't think the current approach takes away from API clarity, as there is no mention of it anywhere, its just the case that all the versioned URLs work if you ping them.

I also messed up the code above your comment, which made the `vMajor` appear in the same level as the optional minor versions. At least *one* versioned URL is still mandatory (here `/vMajor`), which I've separated into a new function.